packages/apps/human-app/frontend/src/modules/worker/email-verification/components/email-verification-process.tsx

```diff
@@ -0,0 +1,28 @@
+import {
+  PageCardError,
+  PageCardLoader,
+} from '@/shared/components/ui/page-card';
+import { getErrorMessageForError } from '@/shared/errors';
+import { useEmailVerification } from '../hooks';
+import { EmailVerificationSuccessMessage } from './email-verification-success-message';
+
+interface EmailVerificationProcessProps {
+  token: string;
+}
+
+export function EmailVerificationProcess({
+  token,
+}: Readonly<EmailVerificationProcessProps>) {
+  const { errorMsg, isEmailVerificationError, isEmailVerificationPending } =
+    useEmailVerification(token);
+
+  if (isEmailVerificationError && errorMsg) {
```

Comment:
What if it's errored, but there is no `errorMsg`? Probably in this case we still should show error

Also you use `getErrorMessageForError` in hook and here, makes sense to keep it only in JSX since `getErrorMessageForError` is purely for user-facing parts (which are components, not hooks)